### DISCUSSION TITLE
Need help with augmented UI (I am a total noob with CSS, HTML and JS)

### DISCUSSION THREAD

**Discussion by TonkatsuRa** [propjockey/augmented-ui#15] [General]
So i tried different ways to get augmented UI to work but I keep failing at it. I even tried using ChatGPT4 to maybe find a way to fix it... but still no chance. 

No matter what I do, the augmented-ui style isn't applied to the container border or any other element of my code. I am 100% sure it is my mistake and I just can't figure out the correct way. The page has full functionality and works well ... just the design is what makes me bite into my table...

Here is my git: https://github.com/TonkatsuRa/Comlink-Thread

**Comment by JaneOri** [comment]
Thanks for checking it out! This should help:

https://github.com/TonkatsuRa/Comlink-Thread/blob/main/comlink.html#L14
and
https://github.com/TonkatsuRa/Comlink-Thread/blob/main/comlink.html#L37

The attribute should be prefixed with `data-`

If you want to use augmented borders on the element, also pass in `border` within the attribute value like:
`data-augmented-ui="border tl-clip br-clip"`
It's best not to use the standard CSS border at the same time.

then to style it:
```css
--aug-border-all: 3px;
--aug-border-bg: #FFFF00;
```

---

Make sure not to import it 3 times when done debugging!
https://github.com/TonkatsuRa/Comlink-Thread/blob/main/comlink.html#L8-L10
:P

----

It's easier to use classes instead but there's two notes here:
https://github.com/TonkatsuRa/Comlink-Thread/blob/main/comlink.css#L195
1) need data- prefix
2) It's not `=` that should be used but `~=` instead (or just use a class name)

----

Lmk if you need anything else!

Thanks again!
// Jane

Oh and ping me on twitter when you're live, I'd love to see the finished work!

**Comment by TonkatsuRa** [comment]
Thank you so much for your reply! I was able to fix my issue so far. The augmented-ui is working now but it threw off everything else. I am currently trying to figure out how to fix it:
Edit: I manage to fix some things. But I can't figure out how to adjust the border color and change the clip styles like you explained on your website. It's driving me completely nuts

If you want, you can have a look: https://codepen.io/TonkatsuRa/pen/VwEMLpo

**Comment by JaneOri** [comment]
If you're doing this in codepen, remove this from your html:
```html
<link rel="stylesheet" type="text/css" href="https://unpkg.com/augmented-ui@2/augmented-ui.min.css">
```

and change the settings for your pen (settings cog in top right of codepen) and import the library from there.:
<img width="526" alt="image" src="https://user-images.githubusercontent.com/7545075/235751162-cb958661-d9c8-4b50-92c6-130251e9a11d.png">

_This IS super annoying (sorry)._

<sub>We didn't have CSS layers or :where() selectors when v2 was released yet so everything in the library is heavy class-weight selectors.
That means the library defaults can override your selectors if you only use a single class and import the library _after_ your custom styles. (which is what happens in codepen if you embed the link tag while using the codepen CSS box)
</sub>

Anyway, after that, the `--aug-border-bg` property will do what you expect:

<img width="497" alt="image" src="https://user-images.githubusercontent.com/7545075/235752147-55762d4e-cdf0-4d8b-b350-e07b6f154f63.png">
<img width="297" alt="image" src="https://user-images.githubusercontent.com/7545075/235752244-be4e183d-12bf-4c6a-9efa-78db66160f1d.png">

<img width="813" alt="image" src="https://user-images.githubusercontent.com/7545075/235752303-69bdae42-3aa0-4c26-bac9-9293df0489fa.png">

**Reply by TonkatsuRa** [reply]
Thank you so much!
I learned a lot and I think I finally start to grasp how augmented-ui works.
The only thing I haven't been able to do, is changing the mixin size of the border cutouts etc.

**Reply by JaneOri** [reply]
looks like you made more progress since then too.

If you inset from the edges with clip, it causes a little pinching. You can fix the pinching like this:
<img width="1263" alt="image" src="https://user-images.githubusercontent.com/7545075/235786145-3b57b1e6-0f46-43b7-8aac-7e2e039ebbd4.png">

<img width="141" alt="image" src="https://user-images.githubusercontent.com/7545075/235786215-a1c5b18a-9eae-4939-8b9f-50c8d9571b18.png">

```css

  --aug-tl-inset1: 35%;
 
  --aug-tr-inset1: 20%;
  --aug-tr1-alt-join-in: initial;
  /* this fixes the pinch ^ that happens when you inset sometimes */

  --aug-tr-inset2: 10%;
  --aug-tr2-alt-join-out: initial;
  /* this fixes the pinch ^ that happens when you inset sometimes  */
  
  --aug-bl-inset2: 10%;

  --aug-br-inset2: 0%;
  
  /* change size of equipped augs */
  --aug-tr1: 10px;
  --aug-tr2: 30px;
```
